File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    // public double sideLen;

    // public Square(double sideLen) {
    //     this.sideLen = sideLen;
    // }
    
    // public double getArea() {
    //     return sideLen * sideLen;
    // }

    private double sideLen;

    public Square(double sideLen){
        this.sideLen = sideLen;
    }

    @Override
    public double getArea(){
        return sideLen * sideLen;
    }

    @Override
    public void draw(){
        System.out.println("Shape printed\nIts area is " + getArea());
    }

    
}

File: typescript/src/shapes/square.ts
// function square(sideLen: number): Shape {
//     return {
//         sideLen,
//         computeArea: function (): number {
//             return sideLen * sideLen
//         }
//     }
// }

import { Shape } from '../shapes/shape';

class Square implements Shape {
    private sideLen: number;

    constructor(sideLen: number) {
        this.sideLen = sideLen;
    }

    getArea(): number {
        return this.sideLen * this.sideLen;
    }

    draw(): void {
        console.log(`Shape printed\nIts area is ${this.getArea()}`);
    }
}

export { Square };



// export { square }